Hey there, thanks for the work you've put in. We're getting this traceback:

henry@metrobiker:~/wps-to-bigcommerce/wps-to-bigcommerce (master)$ python
import.py newesttestingcategory partnumbers.txt
WPS Dealer ID:d2120958
WPS password:
Traceback (most recent call last):
  File "import.py", line 195, in <module>
    part_info = get_part_info(part_number)
  File "import.py", line 73, in get_part_info
    part_response = requests.post(WPS_URL, auth=(DEALER_ID, PASSWORD))
  File "/usr/local/lib/python2.7/dist-packages/requests/api.py", line 109,
in post
    return request('post', url, data=data, json=json, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/api.py", line 50,
in request
    response = session.request(method=method, url=url, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line
465, in request
    resp = self.send(prep, *_send_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line
573, in send
    r = adapter.send(request, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line
415, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', error(110,
'Connection timed out'))
henry@metrobiker:~/wps-to-bigcommerce/wps-to-bigcommerce (master)$

And it created the category "newesttestingcategory" in our test store, but
it didn't add any products.

On Sat, Aug 1, 2015 at 2:26 PM, juanj notifications@github.com wrote:

> I think this solve the problem. Please test it and tell me.
> 
> ## Olizbu
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/151henry151/wps-to-bigcommerce/pull/2
> Commit Summary
> - Fix the broken WPS script
> 
> File Changes
> - _M_ import.py
>   https://github.com/151henry151/wps-to-bigcommerce/pull/2/files#diff-0
>   (104)
> 
> Patch Links:
> - https://github.com/151henry151/wps-to-bigcommerce/pull/2.patch
> - https://github.com/151henry151/wps-to-bigcommerce/pull/2.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/151henry151/wps-to-bigcommerce/pull/2.

## 

Henry Romp
151henry151@gmail.com
M: 917-628-1569
W: 201-223-4344

--- a/import.py
+++ b/import.py
@@ -10,7 +10,7 @@ wps_dealer_id = raw_input('WPS Dealer ID:')
 wps_password = getpass.getpass('WPS password:')
 
 # WPS credentials and path
-WPS_URL = 'https://api.wpswebservices.com'
+WPS_URL = 'http://api.wpswebservices.com'
 DEALER_ID = wps_dealer_id
 PASSWORD = wps_password
 DEBUG = True

--- a/import.py
+++ b/import.py
@@ -32,7 +32,7 @@ OVERWRITE = True
 # Dictionary for converting WPS properties to BigCommerce properties
 conversion_dict = {
     'description': 'name',
-    'list_pric': 'price',
+    'list_price': 'price',
     'estimated_weight': 'weight',
     #'producttype': 'categories',
     'image': 'mainimg',

--- a/import.py
+++ b/import.py
@@ -2,6 +2,7 @@
 
 import requests, json, sys
 import getpass
+from pprint import pprint
 #import urllib3.contrib.pyopenssl
 #urllib3.contrib.pyopenssl.inject_into_urllib3()
 
@@ -27,7 +28,7 @@ BRAND_IMAGE = 'http://162.243.58.11/comingsoon.jpg'
  
 # Script stuff
 OVERWRITE = True
- 
+
 # Dictionary for converting WPS properties to BigCommerce properties
 conversion_dict = {
     'description': 'name',
@@ -70,19 +71,21 @@ def get_part_info(pn):
     part_url = WPS_URL + '/item/%s' % pn
 
     # Send request
-    part_response = requests.post(WPS_URL, auth=(DEALER_ID, PASSWORD))
+    part_response = requests.get(part_url, auth=(DEALER_ID, PASSWORD))
     
     # Convert JSON response and return part info
     try:
         response =  part_response.json()
         if DEBUG:
-            print response
-        return response['response'].get('docs', None)
+            print "[DEBUG] Response of request %s" % part_url
+            print "[DEBUG]", response
+        return response
     
     # If malformed response or error, return None
     except:
         if DEBUG:
-            print part_response.text
+            print "[DEBUG] [ERROR] Error on %s" % part_response.url
+            print "[DEBUG] [ERROR]", part_response.text
         return None
  
 def add_image(image_name, product_id):
@@ -96,7 +99,7 @@ def create_brand(brand_name):
     brand_data = {'name': brand_name, 'image_file': BRAND_IMAGE}
     rp = requests.post(BIG_API + 'brands.json', data=json.dumps(brand_data), headers = BIG_HEADERS, auth=(BIG_USER, BIG_KEY))
     if rp.status_code == 201:
-        print 'Created brand %s' % (brand_name,)
+        print '[LOG] Created brand %s' % (brand_name,)
         b_id =  rp.json()['id']
         brand_dictionary[brand_name] = b_id
         return b_id
@@ -119,7 +122,7 @@ def get_brand_id(name):
 def wps_item_to_bc(wps_item):
     # Swap out compatible keys and remove incompatible keys
     bc_item = {}
-    print "Converting item: %s" % (wps_item['description'],)
+    print "[LOG] Converting item: %s" % (wps_item['description'],)
     for k in wps_item.keys():
         if k in conversion_dict.keys() and k != conversion_dict[k]:
             bc_item[conversion_dict[k]] = wps_item[k]
@@ -133,7 +136,11 @@ def wps_item_to_bc(wps_item):
     # If weight is None , set it to 0
     if not bc_item['weight']:
         bc_item['weight'] = 0
-
+    
+    if DEBUG:
+        print "[DEBUG] Part data: ",
+        pprint(bc_item)
+    
     return bc_item
  
 
@@ -153,22 +160,22 @@ def create_item(pd):
     
     # Check for success
     if rp.status_code == 201:
-        print "Item %s created" % (bigComerce_item['name'],)
+        print "[LOG] Item %s created" % (bigComerce_item['name'],)
         if image_filename:
             product_id = rp.json()['id']
             add_image(image_filename, product_id)
     
     elif rp.status_code == 409 and not OVERWRITE:
-        print "Cannot continue. Item %s already exists" % (bigComerce_item['name'],)
+        print "[LOG] Cannot continue. Item %s already exists" % (bigComerce_item['name'],)
     
     elif rp.status_code == 409 and OVERWRITE:
-        print "Item %s already exists. Overwriting..." % (bigComerce_item['name'],)
+        print "[LOG] Item %s already exists. Overwriting..." % (bigComerce_item['name'],)
         existing_product = requests.get(BIG_STORE_PRODUCT_URL, params={'name': bigComerce_item['name']}, headers = BIG_HEADERS, auth=(BIG_USER, BIG_KEY))
         existing_product = existing_product.json()
         product_id = existing_product[0]['id']
         requests.delete(BIG_API + 'products/%s.json' % (product_id,),
                         headers = BIG_HEADERS, auth = (BIG_USER, BIG_KEY))
-        print 'Deleted item %s' % (bigComerce_item['name'],)
+        print '[LOG] Deleted item %s' % (bigComerce_item['name'],)
         rp = requests.post(BIG_STORE_PRODUCT_URL, data=json.dumps(bigComerce_item), headers = BIG_HEADERS, auth=(BIG_USER, BIG_KEY))
         
         # Get product ID and add image if one exists
@@ -176,10 +183,10 @@ def create_item(pd):
             product_id = rp.json()['id']
             add_image(image_filename, product_id)
         
-        print "Item %s created" % (bigComerce_item['name'],)
+        print "[LOG] Item %s created" % (bigComerce_item['name'],)
     elif DEBUG:
-        print rp.text
-        print 'Could not create item %s' % (bigComerce_item['name'],)
+        print "[DEBUG]", rp.text
+        print '[DEBUG] Could not create item %s' % (bigComerce_item['name'],)
  
 if __name__ == '__main__':
     if len(sys.argv) > 2:
@@ -195,8 +202,8 @@ if __name__ == '__main__':
             part_info = get_part_info(part_number)
             if part_info:
                 if DEBUG:
-                    print "Name: " + part['description']
-                create_item(part)
+                    print "[DEBUG] Name: " + part_info['description']
+                create_item(part_info)
     else:
         print 'Correct syntax is import.py categoryname FILE.TXT\nFILE.TXT is a plain text file with a part number on each line'
  

